lib/Filesystem/Domain/FilePath.php

```diff
@@ -54,19 +57,6 @@ public static function fromUnknown($path): FilePath
         if (is_string($path)) {
             return self::fromString($path);
         }
-
-        if (is_array($path)) {
-            return self::fromParts($path);
-        }
-
-        if ($path instanceof SplFileInfo) {
-            return self::fromSplFileInfo($path);
-        }
-
-        throw new RuntimeException(sprintf(
-            'Do not know how to create FilePath from "%s"',
-            is_object($path) ? get_class($path) : gettype($path)
-        ));
```

Comment:
Those were only used in tests. The rest of the code already uses the correct function.

Reply:
they were used by tests though, removing this changes the purpose of the function (`fromUnknown`)

Reply:
can rename to `fromFilePathOrString` for clarity

Reply:
If possible I would like to remove those kinds of methods entirely and reduce the amount of "unknown"s from phpactor. :D